Panic: runtime error - invalid memory address or nil pointer dereference

When attempting to run the application using go run main.go, a panic occurs due to a runtime error. The error message indicates an invalid memory address or nil pointer dereference. This seems to be related to the database connection setup.
Steps to Reproduce:
Clone the repository.
2. Navigate to the project directory.
Run the application using go run main.go.
Expected Behavior:
The application should connect to the database and run without errors.
Actual Behavior:
The application panics with the following error message:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x100c2fbd4]

goroutine 1 [running]:
github.com/swagisays/karni/karni/global.SetClient(0x100dab788?)
        /Users/test/go/pkg/mod/github.com/swagisays/karni@v0.0.0-20241217025908-cc3c25608810/karni/global/global.go:22 +0xc4
github.com/swagisays/karni/karni.createClient({0x100c39b84, 0x19})
        /Users/test/go/pkg/mod/github.com/swagisays/karni@v0.0.0-20241217025908-cc3c25608810/karni/karni.go:20 +0xfc
github.com/swagisays/karni/karni.Connect({0x100c39b84?, 0x0?}, {0x100c30d67, 0x6})
        /Users/test/go/pkg/mod/github.com/swagisays/karni@v0.0.0-20241217025908-cc3c25608810/karni/karni.go:25 +0x28
main.main()
        /Users/test/Developer/karni_test/main.go:11 +0x34
exit status 2
``` 

Additional Information:

The error seems to originate from the SetClient function in global.go at line 22.
The createClient function in karni.go at line 20 might be passing a nil or improperly initialized value.
The Connect function in karni.go at line 25 is also involved in the call stack.

Environment:

Go version: 1.23.3

Module: karni/test

Potential Solution:

Please investigate the initialization of the client in the SetClient function and ensure that all necessary values are properly initialized before use.

--- a/karni/global/global.go
+++ b/karni/global/global.go
@@ -16,37 +16,53 @@ var (
 	mu            sync.Mutex
 )
 
+func initGlobalContext() {
+	if GlobalContext == nil {
+		GlobalContext = &AppContext{}
+	}
+}
 func SetClient(client *mongo.Client) {
+	initGlobalContext()
 	mu.Lock()
 	defer mu.Unlock()
 	GlobalContext.client = client
 }
 
 func SetDB(db *mongo.Database) {
+	initGlobalContext()
+
 	mu.Lock()
 	defer mu.Unlock()
 	GlobalContext.db = db
 }
 
 func GetClient() *mongo.Client {
+	initGlobalContext()
+
 	mu.Lock()
 	defer mu.Unlock()
 	return GlobalContext.client
 }
 
 func GetDB() *mongo.Database {
+	initGlobalContext()
+
 	mu.Lock()
 	defer mu.Unlock()
 	return GlobalContext.db
 }
 
 func SetGlobalContext(ctx *AppContext) {
+	initGlobalContext()
+
 	mu.Lock()
 	defer mu.Unlock()
 	GlobalContext = ctx
 }
 
 func GetGlobalContext() *AppContext {
+	initGlobalContext()
+
 	mu.Lock()
 	defer mu.Unlock()
 	return GlobalContext

--- a/karni/karni.go
+++ b/karni/karni.go
@@ -24,9 +24,10 @@ func createClient(uri string) (*mongo.Client, error) {
 func Connect(url string, name string) error {
 	client, err := createClient(url)
 	if err != nil {
-		return liberrors.WrapError(1002, "Failed to create MongoDB client", err)
+		return liberrors.WrapError(1001, "Failed to create MongoDB client", err)
 	}
 	db := client.Database(name)
 	global.SetDB(db)
+
 	return nil
 }

--- a/test/main.go
+++ b/test/main.go
@@ -1,0 +1,14 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/swagisays/karni/karni"
+)
+
+func main() {
+	err := karni.Connect("mongodb://localhost:27017", "akjdhaskjdh")
+	if err != nil {
+		fmt.Println(err)
+	}
+}

